backend/app/services/external/aws/s3.py

```
@@ -0,0 +1,62 @@
+                ExpiresIn=expiration,
+            )
+            return {
+                "presigned_url": presigned_url,
+                "key": s3_file_key,
+                "bucket": self.bucket_name,
+            }
+        except ClientError as e:
+            logger.error(f"Error generating pre-signed upload URL: {e}")
+            raise BadRequestException(
+                message=f"Error generating pre-signed upload URL: {e}"
+            )
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)